src/Razor/src/Microsoft.AspNetCore.Razor.Language/Components/ComponentGenericTypePass.cs

```diff
@@ -253,7 +253,7 @@ private void RewriteTypeNames(TypeNameRewriter rewriter, ComponentIntermediateNo
                     else if (childContent.IsParameterized)
                     {
                         // This is a weakly typed parameterized child content, treat it as RenderFragment<object>
-                        childContent.TypeName = ComponentsApi.RenderFragment.FullTypeName + "<System.Object>";
+                        //childContent.TypeName = ComponentsApi.RenderFragment.FullTypeName + "<System.Object>";
```

Comment:
This is not the real fix. This seems to be the culprit though. This fixes the issue and breaks no tests. @rynowak can you help me better understand what the purpose of this line is.

Reply:
I think that this is the bug. I don't know how a child-content could be both weakly-typed and parameterized. 

My guess is that this repro case is where the child content is not generic-typed, and is strongly typed with a parameter `childContent.TypeName != null`. In which case we need to do nothing. So the condition should stay the same as it is, and the comment should be updated. 

So I think we need an extra empty if.